Connection/Connection.php

```diff
@@ -681,21 +683,31 @@ public function read($length, $flags = 0)
             );
         }
 
-        if (true === $this->isEncrypted()) {
-            return fread($this->getStream(), $length);
+        if (is_int($chunk)) {
+            stream_set_chunk_size($this->getStream(), $chunk);
```

Comment:
Should we collect the previous chunk size (returned by `stream_set_chunk_size`) and restore it just after the call? I'm not sure how it acts with concurrent accesses…